tests/e2e/cloudprovider/test-lb-service.sh

```diff
@@ -801,6 +801,160 @@ EOF
     fi
 }
 
+
+########################################################################
+## Name: test_preserve_and_reused_fip
+## Desc: The steps in this test case:
+##   1. Create load balanccer with loadbalancer.openstack.org/keep-floatingip: "true" annotation
+##   2. Delete the load balancer
+##   3. Check if the floating IP is preserved after deletion
+##   4. Create new load balancer reassigning the floating IP allocated in point 1.
+##   5. Delete the service and the allocated floating ip.
+########################################################################
+function test_preserve_and_reused_fip {
+	local service="test-preserve-fip"
+	local second_service="test-reuse-fip"
+
+	printf "\n>>>>>>> Test - preserving Floating IP\n"
+	printf ">>>>>>> Create Service ${service}\n"
+	cat <<EOF | kubectl apply -f -
+kind: Service
+apiVersion: v1
+metadata:
+  name: ${service}
+  namespace: $NAMESPACE
+  annotations:
+    loadbalancer.openstack.org/keep-floatingip: "true"
+spec:
+  type: LoadBalancer
+  selector:
+    run: echoserver
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 8080
+EOF
+
+	printf "\n>>>>>>> Waiting for the Service ${service} creation finished\n"
+	wait_for_service_address ${service}
+	printf "\n>>>>>>> Getting the floating IP of the service\n"
+	FIP=`kubectl -n $NAMESPACE get svc test-preserve-fip --no-headers| awk '{print $4}'`
+	printf "\n>>>>>>> Checking if Floating IP is in openstack"
+	FIP_IN_OS=`openstack floating ip list --floating-ip-address "$FIP" -f json | jq length`
+	if [ "$FIP_IN_OS" -ne "1" ]; then
+	    printf "The floating IP doesn't exist in openstack"
+	    exit 1
+	fi
+
+	printf "\n>>>>>>> Delete Service ${service}\n"
+	kubectl -n $NAMESPACE delete service ${service}
+
+	printf "\n>>>>>>> Checking if Floating IP is still in openstack"
+	FIP_IN_OS=`openstack floating ip list --floating-ip-address "$FIP" -f json | jq length`
```

Comment:
Same here, plus I'd rather rely on ID here to double-check if IP wasn't deleted and recreated.